wagtailorderable/modeladmin/mixins.py

```diff
@@ -1,36 +1,97 @@
-from django.db import transaction
+from django.core.exceptions import FieldDoesNotExist, ImproperlyConfigured, PermissionDenied
+from django.db import connections, transaction
 from django.db.models import F, Count
+from django.db.models.expressions import Case, Value, When
+from django.db.models.functions import Cast
 from django.http.response import HttpResponse, HttpResponseBadRequest
 from django.shortcuts import get_object_or_404
 from django.utils.safestring import mark_safe
 from django.utils.translation import ugettext_lazy as _
+from ..signal import pre_reorder, post_reorder
 
-from wagtailorderable.models import Orderable
+class OrderableMixinMetaClass(type):
+    """
+    index_order method needs to be completed with an `admin_order_field` but as sort_order_field
+    is not yet known in the class, we need this meta class to get it from other final class args
+    """
+    def __new__(cls, name, bases, attrs):
+        model = attrs.get('model', None)
+        sort_order_field = attrs.get('sort_order_field', None)
+        if model and not sort_order_field:
+            sort_order_field = getattr(model, 'sort_order_field', None)
+        if sort_order_field:
+            # unfortunately, wagtail IndexView._get_default_ordering is curently using
+            # `model_admin.ordering` instead of `model_admin.get_ordering()`
+            # So we need to automagically set it here
+            if 'ordering' not in attrs:
+                attrs['ordering'] = (sort_order_field, )
+            elif sort_order_field not in attrs['ordering']:
+                attrs['ordering'] = (sort_order_field, ) + tuple(attrs['ordering'])
+
+            # set the "sorting" column
+            if 'index_order' not in attrs:
+                def index_order(self, obj):
+                    """Content for the `index_order` column"""
+                    return mark_safe((
+                        '<div class="handle icon icon-grip text-replace ui-sortable-handle">'
+                        '%s</div>'
+                    ) % _('Drag'))
+                index_order.admin_order_field = sort_order_field
+                index_order.short_description = _('Order')
+                attrs['index_order'] = index_order
+        return type.__new__(cls, name, bases, attrs)
+
+
+class OrderableMixin(object, metaclass=OrderableMixinMetaClass):
+    sort_order_field = None
 
-class OrderableMixin(object):
     """
     Mixin class to add drag-and-drop ordering support to the ModelAdmin listing
     view when the model extends the `wagtailorderable.models.Orderable`
     abstract model class.
     """
+
     def __init__(self, parent=None):
         super(OrderableMixin, self).__init__(parent)
-        """
-        Don't allow initialisation unless self.model subclasses
-        `wagtail.wagtailcore.models.Orderable`
-        """
-        if not issubclass(self.model, Orderable):
+        # Don't allow initialisation unless self.model subclasses
```

Comment:
not developing a python project for a long time, can you explain why change the comment style here?

Reply:
I don't know either, maybe it was changed by mistake, I've changed it back it in #35